### PR TITLE
Feature/elastic auth tests

### PR DESCRIPTION
**Pull request by bCamba**
**What this PR does / why we need it**:
Added tests for elastic search authentication by api key and by username and password

**Which issue(s) this PR fixes**: #1117 

Please reference the issue this PR will close: #_[issue number]_

**Special notes for your reviewer**: I recently created a PR (#1118) to add api key authentication to elastic search health check. I have now created tests for it and also for the username and password which you did not have tests for. I tried also testing the authentication by certificate but for some reason I was not able to make it work.
Bear in mind that the tests I have created spin up and tear down docker containers automatically so I am not sure you would want those to run in your default CI/CD.

**Does this PR introduce a user-facing change?**: NO

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [x] Unit tests passing
- [x] End-to-end tests passing
- [x] Extended the documentation
- [x] Provided sample for the feature

### CODE REVIEW

File: test/HealthChecks.Elasticsearch.Tests/Fixtures/ElasticContainerFixture.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            _compositeService.Dispose();
+        }
+    }
+
+    public void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+
```
Comment (sungam3r) [inline test/HealthChecks.Elasticsearch.Tests/Fixtures/ElasticContainerFixture.cs:62]:
```suggestion
```

**Review by sungam3r — APPROVED** [review]
OK but I don't use docker so waiting some time for comments from @carlosrecuero .

**Comment by sungam3r** [comment]
@bCamba Formatting check failed

**Comment by sungam3r** [comment]
@bCamba I'm fine to merge after pulling changes from `master`.

**Comment by cieciurm** [comment]
Hi @sungam3r, I could refresh/revamp this PR to make it mergeable.

Could you shortly describe, what should be done within the scope of this PR?

**Comment by sungam3r** [comment]
@cieciurm Just take the code from [bCamba:feature/elastic-auth-tests](https://github.com/bCamba/AspNetCore.Diagnostics.HealthChecks/tree/feature/elastic-auth-tests) branch into your local branch, then merge it with `master` and make **new** PR. Then I can review, merge your PR and close this PR as obsolete.

**Comment by cieciurm** [comment]
Thanks @sungam3r, I already tried that.

The tests against docker are failing and I had hard time trying to fix them.

Should we keep the tests for Authentication (user/password & api key) and wait for someone knowledgeable with integration tests in .NET for Docker or can I just keep the basic tests?

**Comment by sungam3r** [comment]
I have no experience either in Docker or .NET for Docker, ping @bCamba again.

**Comment by cieciurm** [comment]
It's been fixed in #1857

**Pull request closed without merging**